caledonia/src/prehashed.rs

```diff
@@ -0,0 +1,335 @@
+//! Rust implementation of ALBA's prehashed scheme using Blake2b as hash
+//! function, working for big sets.
+
+extern crate core;
+extern crate utils;
+
+use std::f64::consts::E;
+
+/// Setup input parameters
```

Comment:
There's a lot of redundancy between this file and the `bounded.rs` one, would be good to DRY it?

Reply:
No, but factor out the common parts in one or more modules?

Reply:
I had a look at it and while the algo is very similar there are too many tiny differences to properly extract something out (like the DFS). Perhaps we could just remove prehashed? 
I'll still need to keep the current structure for the lottery and weight anyway.

Reply:
Sure, no worries